functions/Resolve-DbaNetworkName.ps1

```diff
@@ -93,175 +87,127 @@ Function Resolve-DbaNetworkName
 						continue
 					}
 				}
-				
-				if ($fqdn -notmatch "\.")
-				{
+
+				if ($fqdn -notmatch "\.") {
 					$dnsdomain = $env:USERDNSDOMAIN.ToLower()
 					$fqdn = "$fqdn.$dnsdomain"
 				}
-				
+
 				$hostname = $fqdn.Split(".")[0]
-				
+
 				[PSCustomObject]@{
-					InputName = $OGComputer
+					InputName    = $OGComputer
 					ComputerName = $hostname.ToUpper()
-					IPAddress = $ipaddress
-					DNSHostname = $hostname
-					Domain = $fqdn.Replace("$hostname.", "")
+					IPAddress    = $ipaddress
+					DNSHostname  = $hostname
+					Domain       = $fqdn.Replace("$hostname.", "")
 					DNSHostEntry = $fqdn
-					FQDN = $fqdn
+					FQDN         = $fqdn
 				}
 				return
 			}
-			
-			Write-Verbose "$functionName - Connecting to $Computer"
-			
-			try
-			{
+
+			Write-Message -Level Verbose -Message "Connecting to $Computer"
+
+			try {
 				$ipaddress = ((Test-Connection -ComputerName $Computer -Count 1 -ErrorAction Stop).Ipv4Address).IPAddressToString
 			}
-			catch
-			{
-				try
-				{
+			catch {
+				try {
 					$ipaddress = ((Test-Connection -ComputerName "$Computer.$env:USERDNSDOMAIN" -Count 1 -ErrorAction SilentlyContinue).Ipv4Address).IPAddressToString
 					$Computer = "$Computer.$env:USERDNSDOMAIN"
 				}
-				catch
-				{
+				catch {
 					$Computer = $OGComputer
 					$ipaddress = ([System.Net.Dns]::GetHostEntry($Computer)).AddressList[0].IPAddressToString
 				}
 			}
-			
-			if ($ipaddress)
-			{
-				Write-Verbose "$functionName - IP Address from $Computer is $ipaddress"
+
+			if ($ipaddress) {
+				Write-Message -Level Verbose -Message "IP Address from $Computer is $ipaddress"
 			}
-			else
-			{
-				Write-Verbose "$functionName - No IP Address returned from $Computer"
-				Write-Verbose "$functionName - Using .NET.Dns to resolve IP Address"
+			else {
+				Write-Message -Level Verbose -Message "No IP Address returned from $Computer"
+				Write-Message -Level Verbose -Message "Using .NET.Dns to resolve IP Address"
 				return (Resolve-DbaNetworkName -ComputerName $Computer -Turbo)
 			}
-			
-			if ($host.Version.Major -gt 2)
-			{
-				Write-Verbose "$functionName - Your PowerShell Version is $($host.Version.Major)"
-				try
-				{
-					Write-Verbose "$functionName - Getting computer information from $Computer via CIM (WSMan)"
-					if ($Credential)
-					{
-						$CIMsession = New-CimSession -ComputerName $Computer -ErrorAction Stop -Credential $Credential
-						$conn = Get-CimInstance -Query "Select * FROM Win32_computersystem" -CimSession $CIMsession -ErrorAction Stop
+
+			if ($host.Version.Major -gt 2) {
+				Write-Message -Level Verbose -Message "Your PowerShell Version is $($host.Version.Major)"
+				try {
+					Write-Message -Level Verbose -Message "Getting computer information from $Computer"
+					if (Was-Bound "Credential") {
+						$conn = Get-DbaCmObject -ClassName win32_ComputerSystem -Computer $Computer -Credential $Credential
```

Comment:
butt hat's not what it's doing - it's allowing throws which is totally diff :/ otherwise, the user just gets warnings but ps cant tell the command didtn execute well